### ISSUE TITLE
Replace puppetlabs-docker_platform with puppetlabs-docker

### ISSUE THREAD

**Issue by hackdna** [refinery-platform/refinery-platform#2727]
[puppetlabs-docker_platform](https://forge.puppet.com/puppetlabs/docker_platform) has been deprecated by its author since October 11, 2017

[puppetlabs-docker](https://forge.puppet.com/puppetlabs/docker) is suggested as its replacement

**Comment by scottx611x** [comment]
I looked into this yesterday I had a bit of trouble initially. I needed to first add the `--verbose` flag [here in bootstrap.sh](https://github.com/refinery-platform/refinery-platform/blob/develop/deployment/bootstrap.sh#L11) to see what was actually going on under the hood.

Updating to the latest [`puppetlabs-docker 1.1.0`](https://forge.puppet.com/puppetlabs/docker) created a chain reaction of conflicting dependencies that I didn't spend much time going down the line of.

In short, `puppetlabs-docker` required newer versions of `puppetlabs-stdlib` & `puppetlabs-apt` that existing puppet modules couldn't utilize.

**Comment by jkmarx** [comment]
Upgrade to puppetlabs/docker 1.0.4. Ref #3118 
Needed to add stahnma/epel package

https://forge.puppet.com/puppetlabs/docker/1.0.4/dependencies
